### PR TITLE
use freebsd built in growfs script

### PR DESCRIPTION
**Pull request by DarkSunOne**
this didn't worked anymore.
tested successfully with the freebsd built in growfs script.
Maybe you want to merge this?

### CODE REVIEW

File: src/etc/rc [hunk 1]
```diff
@@ -73,9 +73,7 @@ while read FS_PART FS_MNT FS_TYPE FS_MORE; do
 		FS_DEV=$(glabel status -as | grep ${FS_DEV} | awk 'match($3, /^[a-z]+[0-9]+/) { print substr( $3, RSTART, RLENGTH )}')
 		if [ "${FS_MNT}" = "/" -a -f ${GROWFS_MARKER} ]; then
 			# hammertime!
-			gpart recover ${FS_DEV}
-			gpart resize -i 1 ${FS_DEV}
-			growfs -y ${FS_MNT}
+			/etc/rc.d/growfs onestart
```
Comment (fichtner) [inline src/etc/rc]: woks for me and also adds ZFS support... yet.. if we go through all the trouble now to do a one-shot script without arguments wouldn't it be better to move it below the fstab loop? just need to check for GROWFS_MARKER this way. :)

Comment (DarkSunOne) [reply src/etc/rc]: Do we have any benefits, moving it outside the loop?
if we let it within the loop, the resize still happens offline, or am i wrong?

I would still prefer to do partition changes with not mounted filesystems ;)

Comment (fichtner) [reply src/etc/rc]: the point is to remove obsolete and unused code

the loop here only reads fstab and configures before mounting the file systems

the next loop will mount the file system

simple solution: use the space in between to execute the growfs code :)

Comment (DarkSunOne) [reply src/etc/rc]: Well, i changed it now, will update the pr when tests are finished.

Is the whole fstab loop (tunefs, trim and so on) still needed?

Comment (fichtner) [reply src/etc/rc]: yes, we need that

Comment (DarkSunOne) [reply src/etc/rc]: files are updated and first test worked without problems :)

File: src/etc/rc [hunk 2]
```diff
@@ -71,12 +71,6 @@ while read FS_PART FS_MNT FS_TYPE FS_MORE; do
 	FS_DEV=$(echo ${FS_PART} | awk 'match($0, /^\/dev\/(gpt|ufs)\/.+$/) { print substr( $0, RSTART + 5, RLENGTH - 5 )}')
 	if [ -n "${FS_DEV}" ]; then
```
Comment (fichtner) [inline src/etc/rc]: zap

File: src/etc/rc [hunk 3]
```diff
@@ -71,12 +71,6 @@ while read FS_PART FS_MNT FS_TYPE FS_MORE; do
...
 		FS_DEV=$(glabel status -as | grep ${FS_DEV} | awk 'match($3, /^[a-z]+[0-9]+/) { print substr( $3, RSTART, RLENGTH )}')
```
Comment (fichtner) [inline src/etc/rc]: zap

File: src/etc/rc [hunk 4]
```diff
@@ -71,12 +71,6 @@ while read FS_PART FS_MNT FS_TYPE FS_MORE; do
...
-		if [ "${FS_MNT}" = "/" -a -f ${GROWFS_MARKER} ]; then
-			# hammertime!
-			gpart recover ${FS_DEV}
-			gpart resize -i 1 ${FS_DEV}
-			growfs -y ${FS_MNT}
-		fi
 	fi
```
Comment (fichtner) [inline src/etc/rc]: zap

File: src/etc/rc [hunk 5]
```diff
@@ -95,6 +89,11 @@ while read FS_PART FS_MNT FS_TYPE FS_MORE; do
 		fi
 	fi
 done < /etc/fstab
+		
+if [ -f ${GROWFS_MARKER} ]; then
+	# hammertime!
```
Comment (fichtner) [inline src/etc/rc]: thanks for retaining this comment ❤️

File: src/etc/rc (hunk 1 above)
Comment (fichtner) [reply src/etc/rc]: RC will be a good opportunity to test this later. Should also work fine on 11.2 from was I can tell. Nice stuff this.

File: src/etc/rc (hunk 5 above)
Comment (DarkSunOne) [reply src/etc/rc]: had to :smile: :heart:

File: src/etc/rc (hunk 2 above)
Comment (DarkSunOne) [reply src/etc/rc]: what do you mean with zap? ^^

Comment (fichtner) [reply src/etc/rc]: all those lines are no longer needed = no code to execute

Comment (fichtner) [reply src/etc/rc]: edited PR, will merge now :)

**Comment by fichtner** [comment]
Merged, thanks! ❤️

**Pull request merged**